webroot/src/models/User/User.model.spec.ts

```diff
@@ -34,24 +82,184 @@ describe('User model', () => {
+                            isWrite: true,
+                            isAdmin: true,
+                        },
+                    ],
+                },
+            ],
+        };
+        const user = new User(data);
+
+        expect(user).to.be.an.instanceof(User);
+        expect(user.permissionsShortDisplay()).to.equal('Admin');
+        expect(user.permissionsFullDisplay()).to.matchPattern([
+            'ASLP: Admin',
+            'Colorado: Write, Admin',
+        ]);
+        expect(user.affiliationDisplay()).to.equal('ASLP');
+        expect(user.statesDisplay()).to.equal('Colorado');
+    });
+    it('should create a User with specific values (state-level permission)', () => {
+        const data = {
+            permissions: [
+                {
+                    compact: new Compact({ type: CompactType.ASLP }),
+                    isRead: false,
+                    isAdmin: false,
+                    states: [
+                        {
+                            state: new State({ abbrev: 'co' }),
+                            isWrite: true,
+                            isAdmin: true,
+                        },
+                    ],
+                },
+            ],
+        };
+        const user = new User(data);
+
+        expect(user).to.be.an.instanceof(User);
+        expect(user.permissionsShortDisplay(CompactType.ASLP)).to.equal('Write, Admin');
+        expect(user.permissionsFullDisplay()).to.matchPattern([
+            'Colorado: Write, Admin',
+        ]);
+        expect(user.affiliationDisplay(CompactType.ASLP)).to.equal('Colorado');
+        expect(user.statesDisplay(CompactType.ASLP)).to.equal('Colorado');
+    });
+    it('should create a User with specific values (state-level permission)', () => {
+        const data = {
+            permissions: [
+                {
+                    compact: new Compact({ type: CompactType.ASLP }),
+                    isRead: false,
+                    isAdmin: false,
+                    states: [
+                        {
+                            state: new State({ abbrev: 'xx' }),
+                            isWrite: false,
+                            isAdmin: true,
+                        },
+                        {
+                            state: new State({ abbrev: 'co' }),
+                            isWrite: false,
+                            isAdmin: true,
+                        },
+                        {
+                            state: new State({ abbrev: 'md' }),
+                            isWrite: false,
+                            isAdmin: true,
+                        },
+                    ],
+                },
+            ],
+        };
+        const user = new User(data);
+
+        expect(user).to.be.an.instanceof(User);
+        expect(user.permissionsShortDisplay(CompactType.ASLP)).to.equal('Admin');
+        expect(user.permissionsFullDisplay()).to.matchPattern([
+            'Unknown: Admin',
```

Comment:
What's the thinking behind keeping a state permission for a non-existent state?

Reply:
As we started displaying permissions with this PR I noticed that IF a state abbreviation were ever to be unknown / unmatched the UI was just displaying zero-length string. I thought that might be less obvious for users & admins to identify data issues, so I updated unmatched states to display "Unknown" - so that it was more obvious to users that there is a gap somewhere.

Reply:
Got it thanks!